### PR TITLE
Fix analysis of big project

### LINKED ISSUE #425: Analysis fails on big project
Analysis of VSCode project fails
```
ERROR: FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
ERROR:  1: node::Abort() [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  4: v8::internal::Factory::NewByteArray(int, v8::internal::PretenureFlag) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  5: v8::internal::SourcePositionTableBuilder::ToSourcePositionTable(v8::internal::Isolate*, v8::internal::Handle<v8::internal::AbstractCode>) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  6: v8::internal::FullCodeGenerator::MakeCode(v8::internal::CompilationInfo*, unsigned long) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  7: v8::internal::FullCodegenCompilationJob::ExecuteJobImpl() [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  8: v8::internal::CompilationJob::ExecuteJob() [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR:  9: v8::internal::(anonymous namespace)::GenerateUnoptimizedCode(v8::internal::CompilationInfo*) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR: 10: v8::internal::(anonymous namespace)::CompileUnoptimizedCode(v8::internal::CompilationInfo*, v8::internal::Compiler::ConcurrencyMode) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR: 11: v8::internal::(anonymous namespace)::GetUnoptimizedCode(v8::internal::CompilationInfo*, v8::internal::Compiler::ConcurrencyMode) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR: 12: v8::internal::Compiler::Compile(v8::internal::Handle<v8::internal::JSFunction>, v8::internal::Compiler::ClearExceptionFlag) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR: 13: v8::internal::Runtime_CompileLazy(int, v8::internal::Object**, v8::internal::Isolate*) [/Users/ssjenka/peach-jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/NodeJS_8.2.1/bin/node]
ERROR: 14: 0x2ca790a040bd
ERROR: 15: 0x2ca790a05470
ERROR: 16: 0x2ca79219cc9d
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 21:53.656s
INFO: Final Memory: 51M/274M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Failed to run external process `node /Users/ssjenka/peach-jenkins/workspace/TypeScript_vscode-osx-6SXK7K3553U5MBU5MBOLT7737SRW3AUJCUSDA7WGYFS435DRTXAA/.sonar/sonarts-bundle/node_modules/tslint-sonarts/bin/tsrunner`. Run with -X for more information
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.executeExternalRunner(ExternalTypescriptSensor.java:237)
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.analyze(ExternalTypescriptSensor.java:133)
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.execute(ExternalTypescriptSensor.java:111)
	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:88)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:82)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:68)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:88)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:180)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:288)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:283)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:261)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:110)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:74)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 2 column 1 path $
	at com.google.gson.Gson.fromJson(Gson.java:891)
	at com.google.gson.Gson.fromJson(Gson.java:817)
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.executeExternalRunner(ExternalTypescriptSensor.java:228)
	... 34 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 2 column 1 path $
	at com.google.gson.stream.JsonReader.beginArray(JsonReader.java:351)
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:70)
	at com.google.gson.Gson.fromJson(Gson.java:879)
	... 36 more
ERROR: 
```

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #425

### CODE REVIEW

File: sonarts-sq-plugin/its/plugin/src/test/java/org/sonar/typescript/its/BigProjectTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  @BeforeClass
+  public static void prepare() {
+    orchestrator.resetData();
+
+    orchestrator.executeBuild(Tests.createScanner("../../../sonarts-core/typescript-test-sources/src/vscode", PROJECT_KEY));
+  }
+
+  @Test
+  public void should_perform_analysis() {
+    assertThat(getProjectMeasureAsDouble("ncloc")).isGreaterThan(0);
```
Comment (stas-vilchik) [inline sonarts-sq-plugin/its/plugin/src/test/java/org/sonar/typescript/its/BigProjectTest.java]: let's check number of issues as well

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/executable/SonarTSCoreBundle.java [hunk 2]
```diff
@@ -87,7 +87,8 @@ private void deploy() {
    */
   @Override
   public SonarTSRunnerCommand getSonarTsRunnerCommand(String tsconfigPath, Iterable<InputFile> inputFiles, TypeScriptRules typeScriptRules) {
-    SonarTSRunnerCommand runnerCommand = new SonarTSRunnerCommand(inputFiles, getNodeExecutable(), this.tsMetricsExecutable.getAbsolutePath());
+    String increaseMemory = "--max-old-space-size=2048";
```
Comment (stas-vilchik) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/executable/SonarTSCoreBundle.java]: extract `2048` to a constant

File: sonarts-core/src/runner/processRequest.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ interface Sensor {
 
 const sensors: Sensor[] = [getHighlighting, getMetrics, getCpdTokens, getSymbolHighlighting];
 
-export function processRequest(inputString: string) {
+export function processRequest(inputString: string): any[] {
```
Comment (stas-vilchik) [inline sonarts-core/src/runner/processRequest.ts]: not needed

File: sonarts-core/tests/runner/sonarUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      callback();
+    },
+  });
+
+  stringifyToStream(stream, data);
+
+  expect(result).toEqual(JSON.stringify(data));
+});
+
+export interface PerFileResult {
```
Comment (stas-vilchik) [inline sonarts-core/tests/runner/sonarUtils.test.ts]: drop it

Comment (vilchik-elena) [reply sonarts-core/tests/runner/sonarUtils.test.ts]: @stas-vilchik I can't, then compilation fails

**Pull request merged**